tests/plugins/test_add_yum_repo_by_url.py

```diff
@@ -169,3 +170,26 @@ def test_multiple_same_repourls_no_suffix(inject_proxy):
     assert workflow.files[os.path.join(YUM_REPOS_DIR, filename)] == repo_content
 
     assert len(workflow.files) == 1
+
+
+def test_invalid_repourl():
+    """Plugin should raise RuntimeError with repo details when invalid URL
+       is used
+    """
+    WRONG_REPO_URL = "http://example.com/nope/repo"
+    tasker, workflow = prepare()
+    runner = PreBuildPluginsRunner(tasker, workflow, [{
+        'name': AddYumRepoByUrlPlugin.key,
+        'args': {'repourls': [WRONG_REPO_URL], 'inject_proxy': None}}])
+
+    (flexmock(YumRepo)
+        .should_receive('fetch')
+        .and_raise(Exception, 'Oh noes, repo is not working!'))
+
+    with pytest.raises(PluginFailedException):
```

Comment:
This isn't our usual idiom for catching a failure in a plugin.

Reply:
I've checked other plugins, did you mean this as proper idiom:
```
with pytest.raises(PluginFailedException) as exc:
    runner.run()
assert '.....' in str(exc)
```
or something else?

Reply:
Updated, please check it now